[Bug] Sort options alphabetically by default

yonikid15 opened the pull request:
🤖 Resolves #9023 

## 👋 Introduction

- Adds default alphabetical sorting to the Combox component.
- Adds prop to turn off sorting if need be.

Question: The Select component has a house build sortingOptions function. I found that [Intl.Collator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator) works well so there's no need to reinvent the wheel. Should I replace `sortOptions` with `alphaSortOptions`?

## 🕵️ Details

Add any additional details that could assist with reviewing or testing this PR.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Run `npm run build`
2. Login as `admin@test.com`
3. Go to any combobox across site and ensure it is sorted alphabetically. For example, the talent requests filters `http://localhost:8000/en/admin/talent-requests`.

## 📸 Screenshot

![Screenshot 2024-01-23 103328](https://github.com/GCTC-NTGC/gc-digital-talent/assets/22059495/0936fdda-dcf5-4ab8-87ad-dd97c2e41c32)

vd1992 commented:
Some quick comments from a quick glance.

Probably want to use the new no sorting prop on this input. 

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/bd177bf1-36c1-4cb0-bcff-8bde91a1b0c0)

Question, do we want to remove the existing sorting functions to avoid redundant sorting? Probably should but this might be scope creep.

Example

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/54b1cb59-b4d8-4563-b1d9-71903327590e)

into

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/1df278ad-dd2c-41b0-9cc6-81808515aa35)

petertgiles reviewed (approved):
Nice, very clean!  Love the tests.

yonikid15 commented:
> Some quick comments from a quick glance.
> 
> Probably want to use the new no sorting prop on this input.

- Nice catch! Fixed by [45ac687](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9066/commits/45ac687dc60ce0096eae25de27486d4840413a56)

> Question, do we want to remove the existing sorting functions to avoid redundant sorting? Probably should but this might be scope creep.

Okay I added a new issue to replace all sorting functions. https://github.com/GCTC-NTGC/gc-digital-talent/issues/9074
